test/finiteness.lean

```diff
@@ -0,0 +1,22 @@
+import Mathlib.Tactic.Finiteness
+import Mathlib.Data.ENNReal.Real
+import Mathlib.MeasureTheory.Measure.Typeclasses
+
+open MeasureTheory
+open scoped ENNReal
+
+example : (1 : ℝ≥0∞) < ∞ := by finiteness
+example : (3 : ℝ≥0∞) ≠ ∞ := by finiteness
+
+example (a : ℝ) (b : ℕ) : ENNReal.ofReal a + b < ∞ := by finiteness
+
+example {a : ℝ≥0∞} (ha : a ≠ ∞) : a + 3 < ∞ := by finiteness
+example {a : ℝ≥0∞} (ha : a < ∞) : a + 3 < ∞ := by finiteness
+example {a : ℝ≥0∞} (ha : a = 0) : a + 3 < ∞ := by finiteness_nonterminal; simp [ha]
```

Comment:
```suggestion
/-- 
Test that `finiteness_nonterminal` makes progress but does not fail on not 
closing the goal. 
-/
example {a : ℝ≥0∞} (ha : a = 0) : a + 3 < ∞ := by finiteness_nonterminal; simp [ha]
```
else I think the point of the test-case isn't entirely clear

Reply:
Done, thanks.